### PR TITLE
Add API spec proposal for user acronym get and delete actions

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR reimplements OpenAPI ReDoc hosting functionality. It also introduces a spec proposal for user acronym get and delete actions. These endpoints will eventually be needed by the frontend in order to acquire and manipulate user's relationships with acronyms.

**Other Changes**
* Update favicon and express route to catch top-level favicon.ico requests
* Mass README documentation update

[SHOWCASE-772](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-772)

## Types of changes

New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

The API proposal can be found here: https://getok-dev.pathfinder.gov.bc.ca/pr-133/api/v1/docs#tag/Users

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-133

File: app/README.md [hunk 1]
```diff
@@ -1,28 +1,229 @@
-# Vue Skeleton Application
+# Common Service Get Token
 
-This node.js skeleton app hosts the Vue Skeleton frontend. It implements a minimal endpoint to allow for Keycloak authentication.
+Common Service Get Token (GETOK) will automate the process for Development teams to get setup to use the common services published on [https://apistore.nrs.gov.bc.ca/store/](https://apistore.nrs.gov.bc.ca/store/).
```
Comment (loneil) [inline app/README.md]: This isn't accurate anymore (https://apistore.nrs.gov.bc.ca/store/ is the WebADE secured ws02 gateway)
Maybe see what Matt wants here?

Comment (jujaga) [reply app/README.md]: I've updated the app and frontend readmes to align with the top-level blurbs.

File: app/frontend/README.md [hunk 2]
```diff
@@ -1,45 +1,110 @@
-# Vue Skeleton Frontend
+# Common Service Get Token
 
-This is the Vue Skeleton frontend. It implements a Vue frontend with Keycloak authentication support.
+This tool will automate the process for getting setup to use common services published on [https://apistore.nrs.gov.bc.ca/store/](https://apistore.nrs.gov.bc.ca/store/).
```
Comment (loneil) [inline app/frontend/README.md]: Same as above

File: app/frontend/README.md [hunk 3]
```diff
@@ -1,45 +1,110 @@
...
 
-## Configuration
+## Frontend
 
-The Vue Skeleton frontend will require some configuration. The API it invokes will be locked down and require a valid JWT Token to access. We will need to configure the application to authenticate using the same Keycloak realm as the [app](../). Note that the Vue Skeleton frontend is currently designed to expect all associated resources to be relative to the original access path.
+The frontend of the application is built on the [Vue.js](https://vuejs.org/) framework as a static UI layer. The web application calls to the GETOK API (see [app](../app)) for authentication and application funtionality.
 
-## Super Quickstart
+The frontend of the application allows users to fill in the appropriate info required to create a WebADE service client and submit it to WebADE, then test token aquisition. The backend API handles communication with the WebADE REST API to create the application configuration. The backend API also creates the password required for the service client and securely returns it to the UI (see [Password Generation](https://github.com/bcgov/nr-get-token/wiki/Password-Generation) for these details)
```
Comment (loneil) [inline app/frontend/README.md]: Also probably want to remove WebADE-specificity

File: app/frontend/README.md (hunk 2 above)
Comment (jujaga) [reply app/frontend/README.md]: I've updated the app and frontend readmes to align with the top-level blurbs.

File: app/frontend/README.md (hunk 3 above)
Comment (jujaga) [reply app/frontend/README.md]: Changed the wording to be more ambiguous

File: app/tests/unit/routes/v1.spec.js [hunk 4]
```diff
@@ -15,7 +15,50 @@ describe(`GET ${basePath}`, () => {
...
+    expect(response.body).toBeTruthy();
+    expect(response.headers['content-type']).toBeTruthy();
+    expect(response.headers['content-type']).toMatch('application/yaml; charset=utf-8');
+    expect(response.text).toContain('openapi: 3.0.2');
+    expect(response.text).toContain('title: GetOK API');
+  });
+});
+
+describe(`GET ${basePath}/api-spec.json`, () => {
+  it('should return the OpenAPI yaml spec', async () => {
```
Comment (TimCsaky) [inline app/tests/unit/routes/v1.spec.js]: shouldnt this read:   it('should return the OpenAPI JSON spec', async () => {

Comment (jujaga) [reply app/tests/unit/routes/v1.spec.js]: I'll fix this in the next pr

**Pull request merged**